tests/time_tests/test_runner/test_timetest.py

```diff
@@ -55,13 +56,13 @@ def test_timetest(instance, executable, niter, cl_cache_dir, test_info, temp_dir
         "device": instance["device"]["name"],
         "niter": niter
     }
-    if exe_args["device"] == "GPU":
-        # Generate cl_cache via additional timetest run
-        _exe_args = exe_args.copy()
-        _exe_args["niter"] = 1
-        logging.info("Run timetest once to generate cl_cache to {}".format(cl_cache_dir))
-        run_timetest(_exe_args, log=logging)
+    logging.info("Run timetest once to generate any cache")
+    retcode, _, _ = run_timetest({**exe_args, "niter": 1}, log=logging)
+    assert retcode == 0, "Run of executable for warm up failed"
+    if cl_cache_dir:
         assert os.listdir(cl_cache_dir), "cl_cache isn't generated"
+    if model_cache_dir:
+        assert os.listdir(model_cache_dir), "model_cache isn't generated"
```

Comment:
Should we hide the range of caches to a single function? assert brings unnecessary duplication.

Reply:
I don't think it's a real issue for the moment